pkg/limitador/redis_storage_options.go

```diff
@@ -26,6 +26,32 @@ func RedisDeploymentOptions(ctx context.Context, cl client.Client, defSecretName
 	}, nil
 }
 
+func DeploymentEnvVar(ctx context.Context, cl client.Client, defSecretNamespace string, configSecretRef *v1.ObjectReference) ([]v1.EnvVar, error) {
+	if configSecretRef == nil {
+		return nil, errors.New("there's no ConfigSecretRef set")
+	}
+
+	_, err := getURLFromRedisSecret(ctx, cl, defSecretNamespace, *configSecretRef)
```

Comment:
I would not call this method, and even remove that method. Let's rely on kubernetes to block the deployment if the secret does not exist.

```go
SecretKeyRef: &v1.SecretKeySelector{
					Key: "URL",
					LocalObjectReference: v1.LocalObjectReference{
						Name: configSecretRef.Name,
					},
                                        Optional: &[]bool{false}[0],  // the deployment should fail if the secret or the key does not exist
				},
```

Reply:
>Let's rely on kubernetes to block the deployment if the secret does not exist.

Wait, what? Where? The `storage` could be disk, which then doesn't require the env var... what am I missing ?

Reply:
That "fake" env var would be added only when storage is redis (or redis-cached). And k8s will require that the secret and the key in the secret exists or the container will not start. 

It is "fake" env var only because it serves secret hiding purposes. Limitador will not read it. It will be the agent deploying the container (the process for simplicity) reading the env var and running limitador with the actual value.

Reply:
But, the status of limitador gives an error if the deployment is not available. So, it should not be necessary. Ref https://github.com/Kuadrant/limitador-operator/blob/main/controllers/limitador_status.go#L111

Furthermore, having a function called `getURLFromRedisSecret` to return a literal like `$(LIMITADOR_OPERATOR_REDIS_URL)` is kind of bad smell. That method is actually doing checking if the secret and the key exists, not even reading the value. 

I will not block this. I am aware that functionally wise it is correct. It just "works". If you thing it is good enough, let me know. I will give the approval after I run the verification steps.

Reply:
I did a bit of looking at this again. I will agree my function name is still not the best and reading the secret without using the value is a bit of a smell. 

What I did find was if you don't do the checks and let the deployment try install you get a `CreateContainerConfigError` when there is no  `URL` key in the secret. There is not error logs and the status blocks are only given minimum replicas and time out error status. I have seen nothing that points to the secret missing the `URL` field. 

@eguzki Which option do you think it is better to have?
1. let k8s fail the deployment because of the missing `URL` in the secret. (clearer operator logic)
2. have the operator block the deployment on missing `URL`  in the secret and give error log messages stating the secret is missing the required fields.

I am thinking option 1 is the better option but it is a bit harder of the user.

Reply:
Good finding. The deployment fails without a good reporting, then I would go for option 2, which is the one you implemented. Add required refactor for a better code if you find appropriate. Maybe a "validation" step before reconciling of any resource happens? Something like in the rate limit policy controller https://github.com/Kuadrant/kuadrant-operator/blob/main/controllers/ratelimitpolicy_controller.go#L154-L162 ???

Reply:
Push a new change. I think it now address this issue/concern.